src/ir/module-splitting.cpp

```diff
@@ -178,13 +197,13 @@ TableSlotManager::TableSlotManager(Module& module)
       if (segmentBase + segment.data.size() >= maxIndex) {
         maxIndex = segmentBase + segment.data.size();
         activeSegment = &segment;
-        activeBase = {"", segmentBase};
+        activeBase = {table.name, "", segmentBase};
       }
     }
   }
   // Initialize funcIndices with the functions already in the table.
   forEachElement(table, [&](Name base, Index offset, Name func) {
-    addSlot(func, {base, offset});
+    addSlot(func, {table.name, base, offset});
   });
```

Comment:
This needs to run for each table in the module. Probably the best way to do that would be to update `forEachElement` to run over each table as well. I think all the places that use `forEachElement` will need to visit every table.

Alternatively you can just assert that there is not more than one table and not implement module splitting for multiple tables right now.